tests/getEditorNamespace.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { getEditorNamespace } from '../src';
+import sinon from 'sinon';
+
+const CKEditorNamespace = window.CKEDITOR;
+
+describe( 'getEditorNamespace', () => {
+	const fakeScriptWithNamespace = 'data:text/javascript;base64,d2luZG93LkNLRURJVE9SID0ge307';
+	const fakeScriptWithoutNamespace = 'data:text/javascript;base64,';
+
+	beforeEach( () => {
+		delete window.CKEDITOR;
+	} );
+
+	afterEach( () => {
+		window.CKEDITOR = CKEditorNamespace;
+	} );
+
+	it( 'should load script and resolve with loaded namespace', () => {
+		return getEditorNamespace( fakeScriptWithNamespace ).then( namespace => {
+			expect( namespace ).to.equal( window.CKEDITOR );
+		} );
+	} );
+
+	it( 'when namespace loaded for the first time should pre-call callback', () => {
+		const spy = sinon.spy();
+		return getEditorNamespace( fakeScriptWithNamespace, spy ).then( namespace => {
+			expect( spy.calledWith( namespace ) ).to.equal( true );
+		} );
+	} );
+
+	it( 'when script doesn\'t provide namespace should reject with an error', () => {
+		return getEditorNamespace( fakeScriptWithoutNamespace ).catch( err => {
+			expect( err ).to.be.a( 'error' );
+			expect( err.message ).to.equal( 'Script loaded from editorUrl doesn\'t provide CKEDITOR namespace.' );
+		} );
+	} );
+
+	it( 'when script cannot be loaded should reject with an error', () => {
```

Comment:
This test is supposed to test the line `41`:
```js
return scriptReject( err);
```
But it happens only if this is the only test. If you run the whole suite, this line isn't executed - the test passes because the other error (from the line `43`) is thrown. Could you take a look why does it happen?

Reply:
Good catch. After a closer look, I see that these tests are written poorly. I've just copied them from Vue and didn't review so deeply - I left with the assumption that if they are on the prod version, they are in good shape. I've rewritten tests and fixed the issue in `getEditorNamespace` where the setting promise to undefined has been set too late.